stack/tools/src/main/java/org/apache/usergrid/tools/ExportAdmins.java

```diff
@@ -60,213 +82,393 @@ public void runTool( CommandLine line ) throws Exception {
+            for (String connectionType : connectionTypes) {
 
-            Results collectionMembers = em.getCollection( entity, collectionName, null, 100000, Level.IDS, false );
+                List<ConnectionRef> connRefs = task.connectionsByType.get( connectionType );
+                if ( connRefs == null ) {
+                    connRefs = new ArrayList<ConnectionRef>();
+                }
 
-            List<UUID> entityIds = collectionMembers.getIds();
+                Results results = em.getConnectedEntities( entity.getUuid(), connectionType, null, Level.IDS );
+                List<ConnectionRef> connections = results.getConnections();
 
-            if ( ( entityIds != null ) && !entityIds.isEmpty() ) {
-                for ( UUID childEntityUUID : entityIds ) {
-                    jg.writeObject( childEntityUUID.toString() );
+                for (ConnectionRef connectionRef : connections) {
+                    connRefs.add( connectionRef );
                 }
             }
+        }
 
-            // End collection array.
-            jg.writeEndArray();
+
+        private void addOrganizationsToTask(AdminUserWriteTask task, Entity entity) throws Exception {
+            task.orgNamesByUuid = managementService.getOrganizationsForAdminUser( entity.getUuid() );
+        }
+
+        public void setDone(boolean done) {
+            this.done = done;
         }
     }
 
+    class AdminUserWriter implements Runnable {
 
-    /**
-     * Persists the connection for this entity.
-     */
-    private void saveDictionaries( Entity entity, EntityManager em, JsonGenerator jg ) throws Exception {
+        private boolean done = false;
+
+        private final BlockingQueue<AdminUserWriteTask> taskQueue;
+
+        public AdminUserWriter( BlockingQueue<AdminUserWriteTask> taskQueue ) {
+            this.taskQueue = taskQueue;
+        }
+
+
+        @Override
+        public void run() {
+            try {
+                writeEntities();
+            } catch (Exception e) {
+                logger.error("Error writing export data", e);
+            }
+        }
+
+
+        private void writeEntities() throws Exception {
+            EntityManager em = emf.getEntityManager( CassandraService.MANAGEMENT_APPLICATION_ID );
 
-        jg.writeFieldName( "dictionaries" );
-        jg.writeStartObject();
+            // write one JSON file for management application users
+            JsonGenerator usersFile =
+                    getJsonGenerator( createOutputFile( ADMIN_USERS_PREFIX, em.getApplication().getName() ) );
+            usersFile.writeStartArray();
 
-        Set<String> dictionaries = em.getDictionaries( entity );
-        for ( String dictionary : dictionaries ) {
+            // write one JSON file for metadata: collections, connections and dictionaries of those users
+            JsonGenerator metadataFile =
+                    getJsonGenerator( createOutputFile( ADMIN_USER_METADATA_PREFIX, em.getApplication().getName() ) );
```

Comment:
So we're using the application's name to name the output file for the admin user metadata? Wouldn't this lead to conflicts in the case that multiple admin users could belong to the same application?